clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -992,6 +997,12 @@ private PartitionRecords parseCompletedFetch(CompletedFetch completedFetch) {
             } else if (error == Errors.TOPIC_AUTHORIZATION_FAILED) {
                 log.warn("Not authorized to read from topic {}.", tp.topic());
                 throw new TopicAuthorizationException(Collections.singleton(tp.topic()));
+            } else if (error == Errors.FENCED_LEADER_EPOCH) {
+                log.warn("Received fenced leader epoch error in fetch for partition {}", tp);
+                this.metadata.requestUpdate();
+            } else if (error == Errors.UNKNOWN_LEADER_EPOCH) {
+                log.warn("Received unknown leader epoch error in fetch for partition {}", tp);
+                this.metadata.requestUpdate();
```

Comment:
I think the metadata update may not be needed. `UNKNOWN_LEADER_EPOCH` means the consumer's metadata has gotten ahead of the broker, so we can just retry. The only thing I am not sure is whether we need additional backoff logic before retrying.